var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -33,34 +55,68 @@ class Flecsi(CMakePackage):
     variant('flecstan', default=False,
             description='Build FleCSI Static Analyzer')
 
-    depends_on('cmake@3.1:',  type='build')
+    depends_on('cmake@3.12:',  type='build')
     # Requires cinch > 1.0 due to cinchlog installation issue
     depends_on('cinch@1.01:', type='build')
     depends_on('mpi', when='backend=mpi')
     depends_on('mpi', when='backend=legion')
-    depends_on('gasnet@2019.3.0 ~pshm', when='backend=legion')
-    depends_on('legion@ctrl-rep +shared +mpi', when='backend=legion')
-    depends_on('boost@1.59.0: cxxstd=11 +program_options')
+    depends_on('mpi', when='backend=hpx')
+    depends_on('openmpi@3.1.3:3.1.4', when='+openmpi')
+    depends_on('mpich@3.2.1', when='+mpich')
+    depends_on('legion@ctrl-rep +shared +mpi +hdf5', when='backend=legion +hdf5')
+    depends_on('legion@ctrl-rep +shared +mpi', when='backend=legion ~hdf5')
+    depends_on('hpx@1.3.0 cxxstd=14', when='backend=hpx')
+    depends_on('boost@1.70.0: cxxstd=14 +program_options')
     depends_on('metis@5.1.0:')
     depends_on('parmetis@4.0.3:')
+    depends_on('hdf5', when='+hdf5')
     depends_on('caliper', when='+caliper')
     depends_on('graphviz', when='+graphviz')
     depends_on('python@3.0:', when='+tutorial')
     depends_on('llvm', when='+flecstan')
 
+    conflicts('+tutorial', when='backend=hpx')
+#    conflicts('+hdf5', when='backend=hpx')
+#    conflicts('+hdf5', when='backend=mpi')
+
     def cmake_args(self):
-        options = ['-DCMAKE_BUILD_TYPE=debug']
-        options.append('-DCINCH_SOURCE_DIR=' + self.spec['cinch'].prefix)
+        spec = self.spec
+        options = ['-DENABLE_MPI=ON',
```

Comment:
Shouldn't this depend on `backend=mpi`?

Reply:
actually all of our backends use mpi so it should always be on.